### PR TITLE
require a project parameter when creating notifications

### PR DESCRIPTION
**Pull request by devoncarew**
- require a `project` parameter when creating Notifications

Without the project parameter, a notification could show up in any open project window (I noticed this when I had ~4 projects open - a message was sent to the wrong window).

Done as two commits for slightly easier review.

### CODE REVIEW

**Review by helin24 — APPROVED** [review]

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 1]
```diff
@@ -54,21 +55,22 @@ public void actionPerformed(@NotNull AnActionEvent event) {
     }
     final String androidStudioPath = findAndroidStudio(event.getProject());
     if (androidStudioPath == null) {
-      FlutterMessages.showError("Unable to locate Android Studio",
-                                "You can configure the Android Studio location via " +
-                                "'flutter config --android-studio-dir path-to-android-studio'.");
+      FlutterMessages.showError(
+        "Unable to locate Android Studio",
+        "You can configure the Android Studio location via 'flutter config --android-studio-dir path-to-android-studio'.",
+        event.getProject());
```
Comment (stevemessick) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java:61]: If enough time has elapsed, `event.getProject()` will return null. It's better to extract it at the beginning of the method, for use later on.

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 2]
```diff
@@ -54,21 +55,22 @@ public void actionPerformed(@NotNull AnActionEvent event) {
...
       return;
     }
 
     final VirtualFile projectFile = findProjectFile(event);
     if (projectFile == null) {
-      FlutterMessages.showError("Error Opening Android Studio", "Project not found.");
+      FlutterMessages.showError("Error Opening Android Studio", "Project not found.", event.getProject());
```
Comment (stevemessick) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java:67]: ditto

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 3]
```diff
@@ -54,21 +55,22 @@ public void actionPerformed(@NotNull AnActionEvent event) {
...
       return;
     }
 
     final VirtualFile file = event.getData(CommonDataKeys.VIRTUAL_FILE);
     final String sourceFile = file == null ? null : file.isDirectory() ? null : file.getPath();
-    openFileInStudio(projectFile, androidStudioPath, sourceFile);
+    openFileInStudio(event.getProject(), projectFile, androidStudioPath, sourceFile);
```
Comment (stevemessick) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java:73]: ditto

File: src/io/flutter/actions/OpenInXcodeAction.java [hunk 4]
```diff
@@ -148,7 +149,7 @@ public void actionPerformed(@NotNull AnActionEvent e) {
       openFile(projectFile);
     }
     else {
-      FlutterMessages.showError("Error Opening Xcode", "Project not found.");
+      FlutterMessages.showError("Error Opening Xcode", "Project not found.", e.getProject());
```
Comment (stevemessick) [inline src/io/flutter/actions/OpenInXcodeAction.java:152]: ditto

File: src/io/flutter/actions/OpenSimulatorAction.java [hunk 5]
```diff
@@ -36,6 +36,6 @@ public void actionPerformed(@NotNull AnActionEvent event) {
     //  }
     //}
 
-    XcodeUtils.openSimulator();
+    XcodeUtils.openSimulator(event.getProject());
```
Comment (stevemessick) [inline src/io/flutter/actions/OpenSimulatorAction.java:39]: ditto

**Review by stevemessick — APPROVED** [review]
I noted a few places where it might be safer to do `event.getProject()` ahead of time.

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 1 above)
Comment (devoncarew) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: Interesting; will do.

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 2 above)
Comment (devoncarew) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: done

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 3 above)
Comment (devoncarew) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: done

File: src/io/flutter/actions/OpenInXcodeAction.java (hunk 4 above)
Comment (devoncarew) [reply src/io/flutter/actions/OpenInXcodeAction.java]: done

File: src/io/flutter/actions/OpenSimulatorAction.java (hunk 5 above)
Comment (devoncarew) [reply src/io/flutter/actions/OpenSimulatorAction.java]: done

**Pull request merged**